scripts/outer/kubectl-kubestellar-prep_for_syncer

```diff
@@ -111,89 +111,67 @@ fi
 	exit 1
 fi
+
 prefixed_stname="$cluster_ns_name"-"$stname"
-stUID=$(kubectl "${kubectl_flags[@]}" get synctargets.edge.kubestellar.io $prefixed_stname -o jsonpath="{.metadata.uid}")
-mbwsname="$spaceID"-mb-"$stUID"
+stUID=$(KUBECONFIG=$espw_space_config kubectl "${kubectl_flags[@]}" get synctargets.edge.kubestellar.io $prefixed_stname -o jsonpath="{.metadata.uid}")
+mbwsname="$imw"-mb-"$stUID"
 
-espw_space_config="${PWD}/temp-space-config/spaceprovider-default-${espw}"
-kubectl-kubestellar-get-config-for-space --space-name ${espw} --provider-name default --sm-core-config $SM_CONFIG --sm-context $SM_CONTEXT --space-config-file $espw_space_config
-if ! KUBECONFIG=$espw_space_config kubectl "${kubectl_flags[@]}" get APIExport edge.kubestellar.io &> /dev/null ; then
+if [ $(KUBECONFIG=$espw_space_config kubectl get crd -l kube-bind.io/exported=true -oname 2>/dev/null | wc -l) -eq 0 ]; then
     echo "$0: it looks like ${espw@Q} is not the edge service provider workspace" >&2
     exit 2
 fi
 
-# TODO: mailbox does not support spaces yet
-if [ "$silent" == "true" ]; then
-    kubectl ws "${kubectl_flags[@]}" "$rootws" > /dev/null
-else
-    kubectl ws "${kubectl_flags[@]}" "$rootws"
-fi
-
-if ! kubectl "${kubectl_flags[@]}" get Workspace "$mbwsname" &> /dev/null; then
+if ! KUBECONFIG=$SM_CONFIG kubectl get space -n spaceprovider-default "$mbwsname" &> /dev/null; then
 	echo "maybe the mailbox controller is slow, give it a chance"
     sleep 15
 fi
 
-if ! kubectl "${kubectl_flags[@]}" get Workspace "$mbwsname" &> /dev/null; then
+if ! KUBECONFIG=$SM_CONFIG kubectl get space -n spaceprovider-default "$mbwsname" &> /dev/null; then
 	echo "maybe the mailbox controller is slow, give it another chance"
     sleep 15
 fi
 
-if ! kubectl "${kubectl_flags[@]}" get Workspace "$mbwsname" &> /dev/null; then
+if ! KUBECONFIG=$SM_CONFIG kubectl get space -n spaceprovider-default "$mbwsname" &> /dev/null; then
     echo "$0: did not find the mailbox workspace $mbwsname; is the mailbox controller running?" >&2
     exit 5
 fi
 
 echo "now that workspace _exists_, but is it _ready_?"
 sleep 5
 
-if ! kubectl ws "${kubectl_flags[@]}" "$mbwsname" &> /dev/null; then
+TESTMB_CMD="KUBECONFIG=$SM_CONFIG kubectl get space -n spaceprovider-default "$mbwsname" -o=jsonpath='{.status.Phase}'"
+
+
+if  phase=($TESTMB_CMD); [ "$phase" != "READY" ]; then
     echo "tried once to switch to $mbwsname"
     sleep 15
-    if ! kubectl ws "${kubectl_flags[@]}" "$mbwsname" &> /dev/null; then
+    if phase=($TESTMB_CMD); [ "$phase" != "READY" ]; then
         echo "tried twice to switch to $mbwsname"
         sleep 15
-        kubectl ws "${kubectl_flags[@]}" "$mbwsname"
     fi
 fi
 
-echo "current workspace is now $(kubectl ws "${kubectl_flags[@]}" . --short)"
-"$bindir/kubectl-kubestellar-syncer_gen" "$prefixed_stname" --syncer-image "$syncer_image" -o "$output"
+mbwsname_space_config="${PWD}/temp-space-config/spaceprovider-default-${mbwsname}"
```

Comment:
"space_config" is more commonly called "kubeconfig", right?
This is not the config of the mailbox space _name_.
I would rename this variable to `mbs_kubeconfig`.